src/Cutscenes.ts

```diff
@@ -1571,6 +1695,115 @@ export class Cutscenes<TEightBittr extends FullScreenPokemon> extends EightBittr
     }
 
     /**
+     *  Cutscene for a Surf attack in battle.
+     * 
+     * @param settings   Settings used for the cutscene.
+     * @param args   Settings for the routine.
+     */
+    public cutsceneBattleAttackSurf(settings: IBattleCutsceneSettings, args: IBattleAttackRoutineSettings): void {
+        const defenderName: string = args.defenderName;
+        const defender: IThing = this.EightBitter.BattleMover.getThing(defenderName) as IThing;
+        const dt: number = 1;
+        const direction: number = defenderName === "opponent" ? -1 : 1;
+        const differenceY: number = defender.height / 3 * this.EightBitter.unitsize;
+        const lineArray: IThing [] = [];
+        const menu: IMenu = this.EightBitter.MenuGrapher.getMenu("BattleDisplayInitial") as IMenu;
+        const wave1: IThing = this.EightBitter.ObjectMaker.make("ExplosionSmall");
+        const wave2: IThing = this.EightBitter.ObjectMaker.make("ExplosionSmall");
+        let dropplets: IThing[] = [this.EightBitter.ObjectMaker.make("ExplosionSmall")];
+        for (let drops: number = 0; drops < 48; drops++) {
+            dropplets.push(this.EightBitter.ObjectMaker.make("ExplosionSmall"));
+        }
+        let startX: number;
+        let startY: number;
+
+        if (direction === -1) {
+            startX = menu.right - defender.width * this.EightBitter.unitsize;
+            startY = menu.top + defender.height * 3;
+        } else {
+            startX = menu.left + defender.width * this.EightBitter.unitsize;
+            startY = menu.bottom - (defender.height + 8) * this.EightBitter.unitsize;
+        }
+        let dropcounter: number = 0;
+        for (let xrain: number = 1; xrain <= 7 ; xrain++) {
+            for (let yrain: number = 1; yrain <= 7; yrain++) {
+                let xdrop: number = menu.left + xrain * menu.width / 2 - yrain * this.EightBitter.unitsize;
+                let ydrop: number = menu.top + yrain * menu.height / 2;
+                this.EightBitter.things.add(dropplets[dropcounter], xdrop, ydrop);
+                dropcounter++;
+            }
+        }
+        this.EightBitter.TimeHandler.addEventInterval (
+            (): void => {
+                // for (const drop of dropplets) {
+                    /**
+                     * Animate the rain
```

Comment:
In general, if you have to add a `todo` like this, also file a corresponding issue and mention the number here.